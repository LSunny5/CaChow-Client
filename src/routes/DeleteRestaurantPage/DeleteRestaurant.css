/*title for delete page*/
.deleteHeading {
    margin: 15px;
}

/*message for delete page*/
.deleteMessage {
    margin: 2rem auto;
    font-size: 1.2em;
}

/*buttons for delete page*/
.deleteButton {
    padding: 0.8rem;
    width: 35%;
}