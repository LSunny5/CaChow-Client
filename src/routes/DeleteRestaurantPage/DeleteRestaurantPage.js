import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import './DeleteRestaurant.css';

export default class DeleteRestaurantPage extends Component {
    render() {
        return (
            <section className='deleteContent'>
                <h2 className="deleteHeading">Deleting Restaurant...</h2>
                <div className="restaurantBox">
                    <p className="deleteMessage">
                        Are you sure you want to delete the restaurant and all the information?
                    </p>
                    <div className = "buttonBox">


                        <NavLink to={'/ownerView'} onClick={this.deleteRestaurant}>Delete</NavLink>
                        <NavLink to={'/ownerView'}>Cancel</NavLink>
                   
                   
                    </div>
                </div>
            </section >
        )
    };
}