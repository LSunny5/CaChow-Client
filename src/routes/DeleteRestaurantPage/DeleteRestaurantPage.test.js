import React from 'react';
import ReactDOM from 'react-dom';
import DeleteRestaurantPage from './DeleteRestaurantPage';
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import { BrowserRouter } from 'react-router-dom';

describe(`DeleteRestaurantPage route`, () => {
    const props = {
        match: {
            params: {
                r_id: 1
            }
        },
        r_id: 1,
        r_owner: 'Owner',
        r_image: 'None',
        r_type: 'Fast',
        r_name: 'Restaurant One',
        r_address: 'Address',
        r_city: 'City',
        r_state: 'State',
        r_zip: 'Zip',
        r_phone: 'Phone',
        r_hours: 1, 
        history: {
            push: () => { }
        }, 
    }

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <BrowserRouter>
                <DeleteRestaurantPage {...props} />
            </BrowserRouter>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the EditRestaurantPage form', () => {
        const wrapper = shallow(<DeleteRestaurantPage {...props} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    });
});