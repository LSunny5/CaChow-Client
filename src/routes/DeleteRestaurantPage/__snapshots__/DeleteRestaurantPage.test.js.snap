// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteRestaurantPage route renders the EditRestaurantPage form 1`] = `
<section
  className="deleteContent"
>
  <h2
    className="deleteHeading"
  >
    Deleting Restaurant...
  </h2>
  <div
    className="restaurantBox"
  >
    <p
      className="deleteMessage"
    >
      Are you sure you want to 
      <span
        className="redandbold"
      >
        delete
      </span>
       the restaurant and all the information?
    </p>
    <div
      className="buttonBox"
    >
      <button
        className="deleteButton redandbold"
        onClick={[Function]}
      >
        Delete
      </button>
      <button
        className="deleteButton"
        onClick={[Function]}
      >
        Cancel
      </button>
    </div>
  </div>
</section>
`;
